### DISCUSSION TITLE
Tutorial fails to document features that need to be enabled.

### DISCUSSION THREAD

**Discussion by furrz** [libp2p/rust-libp2p#3103] [Ideas]
(Edited this several times as I've checked different things and fixed stuff for myself.)

The ping tutorial fails to specify what features ought to be enabled! 

It seems to require, from my testing:

```toml
libp2p = { version = "0.49.0", features = ["ping", "mplex", "noise", "yamux", "websocket", "async-std", "dns", "tcp"] }
```

But the tutorial just puts a bare version number, `libp2p = "0.43.0"`! This is incredibly confusing as a newcomer who has to poke around to figure out what in the world I need to turn on just to make tutorial code compile. Perhaps this ought to be corrected.

**Comment by elenaf9** [comment]
Thanks for reporting @furrz. 
The tutorial specifies libp2p `v0.43`, where we still had the necessary features enabled by default and thus the bare version number was sufficient.

That said, it would be great if you could provide a PR to update it to the most recent version so that other users don't run into the same issue! We provide a `"full"` feature flag which enables all features, so it would be sufficient to just enable that feature.

**Comment by thomaseizinger** [comment]
Related: https://github.com/libp2p/rust-libp2p/issues/3111
